pulsar-functions/runtime/src/main/java/org/apache/pulsar/functions/auth/KubernetesFunctionAuthProvider.java

```diff
@@ -0,0 +1,42 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.pulsar.functions.auth;
+
+import io.kubernetes.client.models.V1ServiceAccount;
+import io.kubernetes.client.models.V1StatefulSet;
+
+/**
+ * Kubernetes runtime specific functions authentication provider
+ */
+public interface KubernetesFunctionAuthProvider extends FunctionAuthProvider {
```

Comment:
It's a bad sign if for the first serious use of an abstraction you need to create another abstraction to work work around shortcomings in the first abstraction. However, I don't think it's even needed in this case. 

The functionAuthData returned by cacheAuthData should contain the actual token. The **caller** of cacheAuthData can then create a secret, and then mount that secret on each of the instance pods. When the instance runs, it can always check for this secret and if it is mounted, the data can be read in and passed to configureAuthenticationConfig.

I don't see why service accounts are needed at all here.

Reply:
> It's a bad sign if for the first serious use of an abstraction you need to create another abstraction to work work around shortcomings in the first abstraction. However, I don't think it's even needed in this case.

Why is that bad?  I designed this so that it is cleaner compared to AuthenticationDataSource where you have a mixture of interfaces to support different authentication methods.  I imagine in the future, different runtimes will have different requirements / interfaces needed to support authentication.  It doesn't make sense to clutter than all together. 

> The functionAuthData returned by cacheAuthData should contain the actual token. 

Why should functionAuthData contain the actual token?  This is implementation specific data.  Different implementations of FunctionAuthenticationProvider should have the flexibility to use it in a way that makes sense for the implementation.

> The caller of cacheAuthData can then create a secret, and then mount that secret on each of the instance pods. 

I think there is some misunderstanding here.  The architecture of Pulsar Function decouples submitting functions and running functions.  The worker that the user actually submits the function do is not necessarily going to be the same worker that is going to run the function.  However, the auth data will only be passed to the worker that the user at first submits his or her function to.  That is why that worker needs to be able to distribute that auth data or data based on that auth data to the rest of the workers that will potentially need to run a function instance.  The interface cacheAuthData returns the data that needs to be distributed to the other workers via the function metadata topic.

Reply:
> I think there is some misunderstanding here. 

Ya, I needed to map out the interactions. See my top level comment.